docs/guides/technical-docs-seo-guide.rst

```diff
@@ -0,0 +1,278 @@
+
+
+Canonical URLs
+~~~~~~~~~~~~~~
+
+Anytime very similar content is hosted at multiple URLs,
+it is pretty important to set a canonical URL.
+The canonical URL tells search engines where the original version
+your documentation is even if you have multiple versions on the internet
+(for example, incomplete translations or deprecated versions).
+
+Read the Docs supports :doc:`setting the canonical URL </guides/canonical>`
+if you are using a :doc:`custom domain </custom_domains>`
+under :guilabel:`Admin` > :guilabel:`Domains`
+in the Read the Docs dashboard.
+
+
+Use a robots.txt file
+~~~~~~~~~~~~~~~~~~~~~
+
+A ``robots.txt`` file is readable by crawlers
+and lives at the root of your site (eg. https://docs.readthedocs.io/robots.txt).
+It tells search engines which pages to crawl or not to crawl
+and can allow you to control how a search engine crawls your site.
+For example, you may want to request that search engines not index
+unsupported versions of your documentation while keeping those docs online in case people need them.
+
+By default, Read the Docs serves a ``robots.txt`` for you
+which points to a :ref:`sitemap <guides/technical-docs-seo-guide:Use a sitemap.xml file>`
+which Read the Docs also generates for you.
+To customize this file, you can create a ``robots.txt`` file
+that is written to your documentation root on your default branch/version.
+
+See the `Google's documentation on robots.txt <https://support.google.com/webmasters/answer/6062608>`_
+for additional details.
+
+
+Use a sitemap.xml file
+~~~~~~~~~~~~~~~~~~~~~~
+
+A sitemap is a file readable by crawlers that contains a list of pages
+and other files on your site and some metadata or relationships about them
+(eg. https://docs.readthedocs.io/sitemap.xml).
+A good sitemaps provides information like how frequently a page or file is updated
+or any alternate language versions of a page.
+
+Read the Docs generates a sitemap for you that contains the last time
+your documentation was updated as well as links to active versions, subprojects, and translations your project has.
+We have a small separate guide on `sitemaps </guides/sitemaps>`_.
+
+See the `Google docs on building a sitemap <https://support.google.com/webmasters/answer/183668>`_.
+
+
+Use meta tags
+~~~~~~~~~~~~~
+
+Using a meta description allows you to customize how your pages
+look in search engine result pages.
+
+Typically search engines will use the first few sentences of a page if no meta description is provided.
+In Sphinx, you can customize your meta description using the following RestructuredText:
+
+.. sourcecode:: rst
+
+    .. meta::
+        :description lang=en:
+            Your meta description here
+
+
+.. figure:: ../_static/images/guides/google-search-engine-results.png
+    :align: center
+    :figwidth: 80%
+    :target: ../_static/images/guides/google-search-engine-results.png
+
+    Google search engine results showing a customized meta description
```

Comment:
This is not a good example :smiley: 

The `description` meta tag of that page is:

```html
<meta content="How to add additional CSS stylesheets or JavaScript files to your Sphinx documentation to override your Sphinx theme or add interactivity with JavaScript." lang="en" name="description" />
```

but Google shows something different for some reason.

This blog post meta description does appear as Google's result description https://blog.readthedocs.com/custom-css-and-js-in-sphinx/

Reply:
You're right. I totally missed that. I wonder why it isn't!

Reply:
It looks like Google chooses whether to use the meta description based on a few factors including the length and whether the keywords in the description match the keywords on the page. Sounds like dark arts.

Reply:
In that case, we may want to mention something "this is black magic and it may not displayed if the engine decides to not show it" :)

Otherwise, I feel that people will come back to us saying: "Hey, meta description does not work on Read the Docs" :stuck_out_tongue:

Reply:
If Google has a guide or the rules for this, we could link to that as well.

Reply:
> Otherwise, I feel that people will come back to us saying: "Hey, meta description does not work on Read the Docs" 😛

I added something to that effect yesterday 😛.

Reply:
Cool! Just saw it. :+1: